packages/orchestration/test/examples/stake-ica.contract.test.ts

```diff
@@ -132,6 +137,56 @@ test('makeAccount, getAddress, getBalances, getBalance', async t => {
   t.snapshot(bootstrap.storage.data.entries(), 'accounts in vstorage');
 });
 
+test('delegate, undelegate, redelegate, withdrawReward', async t => {
+  const { bootstrap } = await commonSetup(t);
+  const { publicFacet } = await startContract(bootstrap);
+  const account = await E(publicFacet).makeAccount();
+
+  // XXX consider building a mock bank into remote chains. for now, assume
+  // newly created accounts magically have tokens.
+  const validatorAddr = {
+    address: 'cosmosvaloper1test' as const,
+    chainId: 'cosmoshub-4',
+    addressEncoding: 'bech32' as const,
+  };
+  const delegation = await E(account).delegate(validatorAddr, {
+    denom: 'uatom',
+    value: 10n,
+  });
+  t.is(delegation, undefined, 'delegation returns void');
+
+  // TODO, fixme!
+  await t.throwsAsync(
+    E(account).undelegate([
+      {
+        shares: '10',
+        validatorAddress: validatorAddr.address,
+      },
+    ]),
+    {
+      message: /bad response/,
+    },
+  );
+
+  const redelgation = await E(account).redelegate(
+    validatorAddr,
+    {
+      ...validatorAddr,
+      address: 'cosmosvaloper2test',
```

Comment:
our composite "address" object reads so weirdly. I hope we can come up with a better name.

Reply:
cc @mitdralla